File: 496-next-greater-element-i/next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        int n=nums1.size();
                int m=nums2.size();
                vector<int> result;


        for(int i=0;i<n;i++){
            bool flag;
            int k,j;
            for( j=0;j<m;j++){
                if(nums1[i]==nums2[j] ){
                    flag=0;
                   
                    for(k=j+1;k<m;k++){
                        if(nums2[j]<nums2[k]){
                            flag=0;
                            break;
                        }
                        if(nums2[j]>=nums2[k]){
                            flag=1;
                        }
                        
                        

                    }
                   
                    break;
                    
                }
                
            }
             if(j==m-1){
                          flag=1;  
                        }
                        if(!flag){
                       result.push_back(nums2[k]); 
                    }
                        
                    if(flag){
                        result.push_back(-1); 
                    }
        }
        return result;
    }
};

File: 496-next-greater-element-i/next-greater-element-i.py
class Solution(object):
    def nextGreaterElement(self, nums1, nums2):
        hashmap={n:i for i,n in enumerate(nums1)}
        result=[-1]*len(nums1)

        for i in range(len(nums2)):
            if nums2[i] not in hashmap:
                continue
            for j in range(i+1,len(nums2)):
                if nums2[j]>nums2[i]:
                    index=hashmap[nums2[i]]
                    result[index]=nums2[j]
                    break 
        return result